### PR TITLE
Adding first unit tests and enabling removal of input `*.gz` extension

### PR DESCRIPTION
**Pull request by MatthiasZepper**
This PR adds the last new feature required for a 0.2 release:

If the output file names are not specified, they are derived from the input file names. For plain input in conjunction with the `-z` / `--gzip` flag, a `*.gz` extension was automatically added, while the other way round had not been implemented yet. This PR introduces removal of a `*.gz` extension on compressed input, if the `-z` / `--gzip` is not specified.

In addition, the first few unit / integration tests have been added. However, a full test coverage will not yet be achievable for this release.

### CODE REVIEW

File: docs/.DS_Store
Comment (maxulysse) [inline docs/.DS_Store:1]: I don't think we need this file

Comment (MatthiasZepper) [reply docs/.DS_Store:1]: Exactly, this is why I am deleting it now. It ended up there by mistake with a previous PR.

Comment (maxulysse) [reply docs/.DS_Store:1]: ohhh, I just noticed that the file was in the PR, not that is was deleted...

:facepalm:

Comment (MatthiasZepper) [reply docs/.DS_Store:1]: No worries, I am very grateful for you taking a look and happy that nothing more serious turned up!

**Pull request merged**